src/v/cluster/data_policy_manager.cc

```diff
@@ -25,27 +26,26 @@ namespace cluster {
 namespace {
 
 template<typename Cmd>
-std::error_code do_apply(Cmd cmd, v8_engine::data_policy_table& db) {
+std::error_code do_apply(Cmd cmd, v8_engine::api& v8_api) {
     return ss::visit(
       std::move(cmd),
-      [&db](create_data_policy_cmd cmd) {
-          return db.insert(cmd.key, std::move(cmd.value.dp))
-                   ? std::error_code(errc::success)
-                   : std::error_code(errc::data_policy_already_exists);
+      [&v8_api](const create_data_policy_cmd& cmd) {
+          return v8_api.insert_data_policy(cmd.key, cmd.value.dp);
       },
-      [&db](delete_data_policy_cmd cmd) {
-          return db.erase(cmd.key)
-                   ? std::error_code(errc::success)
-                   : std::error_code(errc::data_policy_not_exists);
+      [&v8_api](const delete_data_policy_cmd& cmd) {
+          return v8_api.remove_data_policy(cmd.key);
       });
 }
 
 template<typename Cmd>
-ss::future<std::error_code> dispatch_updates_to_cores(
-  Cmd cmd, ss::sharded<v8_engine::data_policy_table>& db) {
-    auto res = co_await db.map_reduce0(
-      [cmd](v8_engine::data_policy_table& local_db) {
-          return do_apply(std::move(cmd), local_db);
+ss::future<std::error_code>
+dispatch_updates_to_cores(Cmd cmd, ss::sharded<v8_engine::api>& v8_api) {
+    if (!v8_api.local_is_initialized()) {
+        co_return std::error_code(cluster::errc::data_policy_not_enabled);
```

Comment:
this is incorrect, what about the cases when data_policy was disabled, we will still have topics created with data policy. This result can not be config dependent.

Reply:
If v8 disabled, user can not set data_policy for topic, because v8_api is not init. And user will get error for incremental alter config with data_policy

Sorry, I do not understand problem a little bit

Reply:
I think that in this case it is ok, sorry for confusion.